app_dart/lib/src/service/scheduler.dart

```diff
@@ -256,3 +257,57 @@ class Scheduler {
     }
   }
 }
+
+/// Load [yamlConfig] to [SchedulerConfig] and validate the dependency graph.
+SchedulerConfig loadSchedulerConfig(YamlMap yamlConfig) {
+  final SchedulerConfig config = SchedulerConfig();
+  config.mergeFromProto3Json(yamlConfig);
+  _validateSchedulerConfig(config);
+
+  return config;
+}
+
+void _validateSchedulerConfig(SchedulerConfig schedulerConfig) {
+  if (schedulerConfig.targets.isEmpty) {
+    throw const FormatException('Scheduler config must have at least 1 target');
+  }
+
+  if (schedulerConfig.enabledBranches.isEmpty) {
+    throw const FormatException('Scheduler config must have at least 1 enabled branch');
+  }
+
+  final Map<String, List<Target>> targetGraph = <String, List<Target>>{};
+  final List<String> exceptions = <String>[];
+  // Construct [targetGraph]. With a one scan approach, cycles in the graph
+  // cannot exist as it only works forward.
+  for (final Target target in schedulerConfig.targets) {
+    if (targetGraph.containsKey(target.name)) {
+      exceptions.add('ERROR: ${target.name} already exists in graph');
+    } else {
+      targetGraph[target.name] = <Target>[];
+      // Add edges
+      if (target.dependencies.isNotEmpty) {
+        if (target.dependencies.length != 1) {
+          exceptions
+              .add('ERROR: ${target.name} has multiple dependencies which is not supported. Use only one dependency');
+        } else {
+          if (target.dependencies.first == target.name) {
+            exceptions.add('ERROR: ${target.name} cannot depend on itself');
+          } else if (targetGraph.containsKey(target.dependencies.first)) {
+            targetGraph[target.dependencies.first].add(target);
```

Comment:
do we need to maintain the child list of a target?

Reply:
This graph is specifically for validating configs. For future scheduling, we'll want to use a hash map for tracking what builds should be triggered when. So when `$target` notifies of completion, we call `triggers[$target]` to quickly lookup the next set.
```
trigger['web_engine_linux'] = <Target>['web_engine_linux_chrome', 'web_engine_linux_firefox', ...];
```